parsimonious/expressions.py

```diff
@@ -224,9 +223,18 @@ def __init__(self, literal, name=''):
         super(Literal, self).__init__(name)
         self.literal = literal
 
+    def __hash__(self):
```

Comment:
No, just in `GrammarVisitor`. It _could_ be a hotspot in some code that calls it if that code dynamically constructs large numbers of grammars in some programmatic fashion. I could imagine applications which might want to do that, but it's probably best to optimize those applications if/when they come up.